test/parallel/test-https-server-keep-alive-timeout.js

```diff
@@ -58,32 +55,22 @@ test(function serverKeepAliveTimeoutWithPipeline(cb) {
       c.write('GET /2 HTTP/1.1\r\nHost: localhost\r\n\r\n');
       c.write('GET /3 HTTP/1.1\r\nHost: localhost\r\n\r\n');
     });
-    const interval = setInterval(() => {
-      if (socket && socket.destroyed && requestCount === 3) {
-        clearInterval(interval);
-        server.close();
-        cb();
-      }
-    }, 1);
   }));
 });
 
 test(function serverNoEndKeepAliveTimeoutWithPipeline(cb) {
-  let socket;
-  let timeoutCount = 0;
   let requestCount = 0;
   process.on('exit', () => {
-    assert.strictEqual(timeoutCount, 1);
     assert.strictEqual(requestCount, 3);
   });
   const server = https.createServer(serverOptions, (req, res) => {
-    socket = req.socket;
     requestCount++;
   });
-  server.setTimeout(500, (socket) => {
-    timeoutCount++;
+  server.setTimeout(500, common.mustCall((socket) => {
     socket.destroy();
-  });
+    server.close();
```

Comment:
nit: just for readability, could you put comment that `// this ends the test` or do a:
```js
function endTest(socket) {
  socket.destroy();
  server.close();
  cb();
}
...
server.setTimeout(500, common.mustCall(endTest);
```
Here and in L41?